mpu6050/mpu6050.go

```diff
@@ -1,95 +1,251 @@
+	// setClockSource
+	var pwrMgt [1]byte
+
+	if err = p.read(PWR_MGMT_1, pwrMgt[:]); err != nil {
+		return err
+	}
+	pwrMgt[0] = (pwrMgt[0] & (^CLK_SEL_MASK)) | clkSel // Escribo solo el campo de clk_sel
+	if err = p.write8(PWR_MGMT_1, pwrMgt[0]); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetGyroRange configures the full scale range of the gyroscope.
+// It has four possible values +- 250°/s, 500°/s, 1000°/s, 2000°/s.
+// The function takes values of gyroRange from 0-3 where 0 means the
+// lowest FSR (250°/s) and 3 is the highest FSR (2000°/s).
+func (p *Dev) SetGyroRange(gyroRange byte) (err error) {
+	switch gyroRange {
+	case GYRO_RANGE_250:
+		p.gRange = 250
+	case GYRO_RANGE_500:
+		p.gRange = 500
+	case GYRO_RANGE_1000:
+		p.gRange = 1000
+	case GYRO_RANGE_2000:
+		p.gRange = 2000
+	default:
+		return fmt.Errorf("invalid gyroscope FSR input")
+	}
+	// setFullScaleGyroRange
+	var gConfig [1]byte
+
+	if err = p.read(GYRO_CONFIG, gConfig[:]); err != nil {
+		return err
+	}
+	gConfig[0] = (gConfig[0] & (^G_FS_SEL)) | (gyroRange << G_FS_SHIFT) // Escribo solo el campo de FS_sel
+
+	if err = p.write8(GYRO_CONFIG, gConfig[0]); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetAccelRange configures the full scale range of the accelerometer.
+// It has four possible values +- 2g, 4g, 8g, 16g.
+// The function takes values of accRange from 0-3 where 0 means the
+// lowest FSR (2g) and 3 is the highest FSR (16g)
+func (p *Dev) SetAccelRange(accRange byte) (err error) {
+	switch accRange {
+	case ACCEL_RANGE_2:
+		p.aRange = 2
+	case ACCEL_RANGE_4:
+		p.aRange = 4
+	case ACCEL_RANGE_8:
+		p.aRange = 8
+	case ACCEL_RANGE_16:
+		p.aRange = 16
+	default:
+		return fmt.Errorf("invalid accelerometer FSR input")
+	}
+	// setFullScaleAccelRange
+	var aConfig [1]byte
+
+	if err = p.read(ACCEL_CONFIG, aConfig[:]); err != nil {
+		return err
+	}
+	aConfig[0] = (aConfig[0] & (^AFS_SEL)) | (accRange << AFS_SHIFT) // Escribo solo el campo de FS_sel
+
+	if err = p.write8(ACCEL_CONFIG, aConfig[0]); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetSleep sets the sleep bit on the power managment 1 field.
+// When the recieved bool is true, it sets the bit to 1 thus putting
+// the peripheral in sleep mode.
+// When false is recieved the bit is set to 0 and the peripheral wakes
+// up.
+func (p *Dev) SetSleep(sleep bool) (err error) {
+	// setSleepBit
+	var pwrMgt [1]byte
+
+	if err = p.read(PWR_MGMT_1, pwrMgt[:]); err != nil {
+		return err
+	}
+	if sleep {
+		pwrMgt[0] = (pwrMgt[0] & (^SLEEP_MASK)) | (1 << SLEEP_SHIFT) // Escribo solo el campo de clk_sel
+	} else {
+		pwrMgt[0] = (pwrMgt[0] & (^SLEEP_MASK))
+	}
+	//Envio el registro modificado al periferico
+	if err = p.write8(PWR_MGMT_1, pwrMgt[0]); err != nil {
+		return err
+	}
+	return nil
+}
+
+func DefaultConfig() IMUConfig {
+	return IMUConfig{
+		AccRange:    ACCEL_RANGE_16,
+		GyroRange:   GYRO_RANGE_2000,
+		sampleRatio: 0, // TODO add const values.
+		clkSel:      0,
+	}
```

Comment:
Any reason not to use the zero value for these options as the default value?

Reply:
I'm not sure I understand- I've added them here redundantly just so I can mark it with a TODO